### PR TITLE
Fix `yarn build` for Windows

### LINKED ISSUE #8890: [BUG] Unable to `yarn build` on Windows
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Willingness to contribute

Yes. I can contribute a fix for this bug independently.

### MLflow version

- Client: 2.4.1
- Tracking server: 2.4.1

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 11
- **Python version**: 3.9.7
- **yarn version, if running the dev UI**: 3.5.0

### Describe the problem

Running `yarn build` for dev UI fails on Windows due to regex path matching in craco config:

https://github.com/mlflow/mlflow/blob/221d0f4a5c15f3a10dbc3b66c7a064c5b6091f2d/mlflow/server/js/craco.config.js#L73

We might need to change to something like `.match(/[\/\\]mini-css-extract-plugin[\/\\]/` ?

### Tracking information

```shell
REPLACE_ME
```

### Code to reproduce issue

```shell
$ cd mlflow/server/js
$ yarn build
```

### Stack trace

```shell
craco:  *** Cannot find ESLint plugin (ESLintWebpackPlugin). ***
C:\Users\gabrielfu\PycharmProjects\mlflow\mlflow\server\js\craco.config.js:82
    throw new Error('Failed to fix CSS paths!');
          ^

Error: Failed to fix CSS paths!
    at configureIframeCSSPublicPaths (C:\Users\gabrielfu\PycharmProjects\mlflow\mlflow\server\js\craco.config.js:82:11)
    at configure (C:\Users\gabrielfu\PycharmProjects\mlflow\mlflow\server\js\craco.config.js:295:25)
    at giveTotalControl (C:\Users\gabrielfu\PycharmProjects\mlflow\mlflow\server\js\node_modules\@craco\craco\lib\features\webpack\merge-webpack-config.js:57:25)
    at mergeWebpackConfig (C:\Users\gabrielfu\PycharmProjects\mlflow\mlflow\server\js\node_modules\@craco\craco\lib\features\webpack\merge-webpack-config.js:106:38)
    at overrideWebpackProd (C:\Users\gabrielfu\PycharmProjects\mlflow\mlflow\server\js\node_modules\@craco\craco\lib\features\webpack\override.js:18:36)
    at C:\Users\gabrielfu\PycharmProjects\mlflow\mlflow\server\js\node_modules\@craco\craco\scripts\build.js:26:5
```

### Other info / logs

```
REPLACE_ME
```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [X] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [X] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by gabrielfu**
Signed-off-by: Gabriel Fu <hfu.gabriel@gmail.com>

## Related Issues/PRs

resolves #8890 

## What changes are proposed in this pull request?

use platform separator in regex matching

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [X] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/server/js/craco.config.js [hunk 1]
```diff
@@ -70,7 +70,7 @@ function configureIframeCSSPublicPaths(config, env) {
         .filter((oneOf) => oneOf.test?.toString() === /\.css$/.toString())
         .forEach((cssRule) => {
           cssRule.use
-            ?.filter((loaderConfig) => loaderConfig?.loader.match(/\/mini-css-extract-plugin\//))
+            ?.filter((loaderConfig) => loaderConfig?.loader.match(new RegExp(`\\${path.sep}mini-css-extract-plugin\\${path.sep}`)))
```
Comment (harupy) [inline mlflow/server/js/craco.config.js:73]:
```
            ?.filter((loaderConfig) => loaderConfig?.loader.split(path.sep).some(p => p === "mini-css-extract-plugin"))
```

Can we split by `path.sep` and find `"mini-css-extract-plugin"`?

Comment (gabrielfu) [reply mlflow/server/js/craco.config.js:73]: sure that works too, I'll slice the split to make sure `mini-css-extract-plugin` is an intermediate folder

File: mlflow/server/js/craco.config.js [hunk 2]
```diff
@@ -70,7 +70,8 @@ function configureIframeCSSPublicPaths(config, env) {
         .filter((oneOf) => oneOf.test?.toString() === /\.css$/.toString())
         .forEach((cssRule) => {
           cssRule.use
-            ?.filter((loaderConfig) => loaderConfig?.loader.match(/\/mini-css-extract-plugin\//))
+            // matches "/mini-css-extract-plugin/"
+            ?.filter((loaderConfig) => loaderConfig?.loader.split(path.sep).slice(1, -1).some(p => p == "mini-css-extract-plugin"))
```
Comment (harupy) [inline mlflow/server/js/craco.config.js:74]: Can we just use `[\\/]`?

```js
?.filter((loaderConfig) => loaderConfig?.loader.match(/[\\/]mini-css-extract-plugin[\\/]/))
```

Comment (gabrielfu) [reply mlflow/server/js/craco.config.js:74]: sure

**Review by harupy — APPROVED** [review]
LGTM!

**Pull request merged**